packages/console/src/pages/SignInExperience/utilities.ts

```diff
@@ -57,11 +57,19 @@ export const signInExperienceParser = {
   },
   toRemoteModel: (setup: SignInExperienceForm): SignInExperience => {
     const {
+      branding,
       languageInfo: { mode, fallbackLanguage, fixedLanguage },
     } = setup;
 
     return {
       ...setup,
+      branding: {
+        ...branding,
+        // Transform empty string to undefined
+        darkPrimaryColor: branding.darkPrimaryColor?.length ? branding.darkPrimaryColor : undefined,
+        darkLogoUrl: branding.darkLogoUrl?.length ? branding.darkLogoUrl : undefined,
+        slogan: branding.slogan?.length ? branding.slogan : undefined,
```

Comment:
Why do we want to check the length first? Would it be equivelant to just
```typescript
const { darkPrimaryColor, darkLogoUrl, slogan } = branding ?? {};

return {
  ...setup,
  branding: { ...branding, darkPrimaryColor, darkLogoUrl, slogan },
}
```

Reply:
to remove empty string `''`, or it won't pass zod check in API.

Reply:
a `null` or `undefined` value applied to our `input` will become `''`

Reply:
the solution is ugly though, any idea?